mongodb-kubernetes-operator.yaml

```diff
@@ -47,6 +46,76 @@ subpackages:
             stat /manager
             stat /usr/local/bin/entrypoint
 
+  # Note: Upstream uses different versioning for components (e.g., operator: 0.12.x, version hook: 1.0.x),
+  # but Chainguard images use a unified tag matching the main operator version due to Git-based automation.
+  # See upstream version mapping in release.json: https://github.com/mongodb/mongodb-kubernetes-operator/blob/master/release/release.json
+  # All components are built from the same source, even if upstream lacks separate tags.
+  - name: ${{package.name}}-version-upgrade-post-start-hook
+    dependencies:
+      runtime:
+        - coreutils
+        - bash
+    pipeline:
+      - uses: go/build
+        with:
+          output: version-upgrade-hook
+          packages: ./cmd/versionhook/main.go
+    test:
+      environment:
+        environment:
+          AGENT_STATUS_FILEPATH: "/var/log/mongodb-mms-automation/healthstatus/agent-health-status.json"
+      pipeline:
+        - name: "Verify version-upgrade-hook"
+          uses: test/daemon-check-output
+          with:
+            start: |
+              version-upgrade-hook
+            timeout: 60
+            expected_output: |
+              Running version change post-start hook
+
+  - name: ${{package.name}}-version-upgrade-post-start-hook-compat
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.subpkgdir}}/
+          ln -sf /usr/bin/version-upgrade-hook ${{targets.subpkgdir}}/version-upgrade-hook
+    test:
+      pipeline:
+        - runs: |
+            stat /version-upgrade-hook
+
+  - name: mongodb-kubernetes-readinessprobe
```

Comment:
that'd be great if we could rename the sub-package as ${{package.name}}-readinessprobe smth like that.